server/src/logger.ts

```diff
@@ -0,0 +1,153 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript/lib/tsserverlibrary'; // used as value
+
+// NOTES:
+// Be very careful about logging. There are two types of logging:
+// 1. Console
+// 2. File
+// The language server could operate in a few different modes, depending on
+// startup options. By default, it communicates with the client through
+// Node IPC. In this case, logging to stdout/stderr would be piped to VSCode's
+// development console. Verbose log should not be sent to the client as it
```

Comment:
Is the dev console sent to the client?

Reply:
The dev console is the "Output" tab in VSCode, so, yeah, it is sent to client

Reply:
Can you clarify this in the comments?